web-common/src/features/dashboards/state-managers/actions/types.ts

```diff
@@ -26,14 +26,25 @@ export type DashboardCallbackExecutor = (
   callback: DashboardMutatorCallback
 ) => void;
 
+/**
+ * DashboardMutatorFnGeneralArgs are the arguments that are passed
+ * to all DashboardMutatorFn functions. This includes a reference
+ * to the live dashboard, and a callback that can be used to cancel
+ * queries if needed.
+ */
+export type DashboardMutatorFnGeneralArgs = {
+  dashboard: MetricsExplorerEntity;
+  cancelQueries: () => void;
+};
```

Comment:
This type above expands what is passed into actions, enabling them to cancel queries. Can pass additional functionality in the future by adding things here